course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -74,61 +56,632 @@ describe('Unit testing PeopleController', function() {
+            // value from scope.selectedRole.roleName, but doesn't hurt.
+            expect(modalScope.selectedRoleName)
+                .toEqual(scope.selectedRole.roleName);
+        });
+
+        it('initiates the process if confirmed', function() {
+            scope.confirmAddModalInstance.close();
+            scope.$digest();  // resolves modalInstance result
+            expect(scope.addPeopleToCourse).toHaveBeenCalled();
+        });
+
+        it('doesn\'t initiate the process if canceled', function() {
+            scope.confirmAddModalInstance.dismiss();
+            scope.$digest();  // resolves confirmAddModalInstance result
+            expect(scope.addPeopleToCourse).not.toHaveBeenCalled();
+        });
+
     });
     describe('getMembersByUserId', function() {
-        it('maps enrollments to IDs');
-        it('patches/hacks/fixes up the Teaching Fellow role name');
+        beforeEach(function() {
+            controller = $controller('PeopleController', {$scope: scope });
+        });
+        afterEach(function() {
+            // handle the course instance get from setTitle, so we can always
+            // assert at the end of a test that there's no pending http calls.
+            $httpBackend.expectGET(courseInstanceURL).respond(200, '');
+            $httpBackend.flush(1);
+        });
+        it('maps enrollments to IDs', function() {
+            var memberList = [{
+                    "role": {
+                        "canvas_role": "student",
+                        "role_id": 101,
+                        "role_name": "Teaching Fellow"
+                    },
+                    "source": "xmlfeed",
+                    "user_id": "12345678",
+                    "profile": {
+                        "active": 1,
+                        "email_address": "smith@harvard.edu",
+                        "name_first": "Sarah",
+                        "name_last": "Smith",
+                        "prime_role_indicator": "Y",
+                        "role_type_cd": "STUDENT",
+                        "univ_id": "12345678"
+                    }
+                },
+                {
+                    "role": {
+                        "canvas_role": "student",
+                        "role_id": 102,
+                        "role_name": "TA"
+                    },
+                    "source": "xmlfeed",
+                    "user_id": "87654321",
+                    "profile": {
+                        "active": 1,
+                        "email_address": "smith@harvard.edu",
+                        "name_first": "kid",
+                        "name_last": "smart",
+                        "prime_role_indicator": "Y",
+                        "role_type_cd": "STUDENT",
+                        "univ_id": "87654321"
+                    }
+                }];
+            var result = scope.getMembersByUserId(memberList);
+            memberList[0].role.role_name = 'TA';
```

Comment:
To be explicit, might be nicer to copy `memberList[0]` into a new object, update the `role.role_name` there, and compare with that.